src/Compilers/Test/Core/Diagnostics/DiagnosticDescription.cs

```diff
@@ -51,14 +53,16 @@ private IEnumerable<string> GetArgumentsAsStrings()
             {
                 // We'll use IFormattable here, because it is more explicit than just calling .ToString()
                 // (and is closer to what the compiler actually does when displaying error messages)
-                _argumentsAsStrings = _arguments.Select(o =>
+                _argumentsAsStrings = _arguments.Select((o, i) =>
                 {
                     if (o is DiagnosticInfo embedded)
                     {
                         return embedded.GetMessage(EnsureEnglishUICulture.PreferredOrNull);
                     }
 
-                    return string.Format(EnsureEnglishUICulture.PreferredOrNull, "{0}", o);
+                    return !_originalFormatSpecifiers.IsDefault && i < _originalFormatSpecifiers.Length ?
+                        string.Format(_originalFormatSpecifiers[i], o) :
+                        string.Format(EnsureEnglishUICulture.PreferredOrNull, "{0}", o);
```

Comment:
>string.Format(EnsureEnglishUICulture.PreferredOrNull, "{0}", o) [](http://example.com/codeflow?start=24&length=63)

Do we hit this case, or does `_originalFormatSpecifiers.Length == _arguments.Length` always?

Reply:
Yes, because of diagnostics `ERR_IdentifierExpectedKW`, so we have to handle a possible mismatch